### PR TITLE
Bug 1168185 - Implement the skeleton of Data Sync in Settings app.

### PR DESCRIPTION
**Pull request by weilonge**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a4b837dc361f6c83682f0d5ea9f09014209b91f5

File: apps/settings/elements/fxsync.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    <gaia-header action="back" data-href="#root">
+      <h1 data-l10n-id="fxsync"></h1>
+    </gaia-header>
+
+    <div>
+      <div id="fxsync-signin">
+        <ul>
+          <li id="fxsync-enabled">
+            <label class="pack-switch">
```
Comment (gasolin) [inline apps/settings/elements/fxsync.html]: please use `gaia-switch` instead once its going to review stage :)

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d7a6d916734e74035b3c5c76450d7bf3e0b362be

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2c53f1174bd2bb45865f7d916f793083486215be

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=43194f673bd932e70d67ba2884a3e8171f15fbdf

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8d830ecdf2e913bf672674fa0084b5ee473b2737

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=963c3875d5dcd274471e66a2336f2a81de24ecc0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4ac77b1a85acf5b09fe8856235bfb2697167a82e

File: apps/settings/js/fxsync.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      // shrink it. Bug 1087441
+      var header = document.querySelector('#fxsync gaia-header');
+      SettingsUtils.runHeaderFontFit(header);
+    },
+  };
+
+  return FxSync;
+});
+
+navigator.mozL10n.once(function() {
```
Comment (evelynhung) [inline apps/settings/js/fxsync.js]: According to bug 1187668, we don't have to wait `navigator.mozL10n.once` ready anymore.

File: apps/settings/js/fxsync.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+define('fxsync', ['modules/settings_utils', 'shared/settings_listener'
+], function(SettingsUtils, SettingsListener) {
```
Comment (evelynhung) [inline apps/settings/js/fxsync.js]: do you listen to any settings change?

File: apps/settings/elements/fxsync.html [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        </ul>
+      </div>
+
+      <div id="fxsync-collections">
+        <header>
+          <h2 data-l10n-id="fxsync-collections">Collections</h2>
+        </header>
+        <ul>
+          <li>
+            <gaia-switch>
```
Comment (evelynhung) [inline apps/settings/elements/fxsync.html]: I didn't see you add new settings key, is it intentional? If so, could you explain more how the settings will be set?

File: apps/settings/elements/fxsync.html [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+              <label data-l10n-id="fxsync-collections-bookmarks">Bookmarks</label>
+              <details></details>
+            </gaia-switch>
+          </li>
+          <li>
+            <gaia-switch>
+              <label data-l10n-id="fxsync-collections-history">History</label>
+              <details></details>
```
Comment (evelynhung) [inline apps/settings/elements/fxsync.html]: same here.

File: apps/settings/elements/fxsync.html [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        </header>
+        <ul>
+          <li>
+            <gaia-switch>
+              <label data-l10n-id="fxsync-collections-bookmarks">Bookmarks</label>
+              <details></details>
+            </gaia-switch>
+          </li>
+          <li>
+            <gaia-switch>
```
Comment (evelynhung) [inline apps/settings/elements/fxsync.html]: same here

File: apps/settings/elements/fxsync.html [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+    <gaia-header action="back" data-href="#root">
+      <h1 data-l10n-id="fxsync"></h1>
+    </gaia-header>
+
+    <div>
+      <div id="fxsync-signin">
+        <ul>
+          <li id="fxsync-enabled">
+            <gaia-switch>
```
Comment (evelynhung) [inline apps/settings/elements/fxsync.html]: same below

File: apps/settings/elements/fxsync.html [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+              <label data-l10n-id="fxsync">Firefox Sync</label>
+              <details></details>
+            </gaia-switch>
+            <p class="explanation">Signed-in:<br><span id="sync-account"></span></p>
+          </li>
+          <li data-state="ready">
+            <button id="sync-button" data-l10n-id="fxsync-sync-now"></button>
+            <p class="explanation">Last Sync Time:<br><span id="sync-time"></span></p>
```
Comment (evelynhung) [inline apps/settings/elements/fxsync.html]: How do we get the sync time updated? I didn't fine this logic in your JS file.

File: apps/settings/elements/fxsync.html [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+            </gaia-switch>
+          </li>
+          <li>
+            <gaia-switch>
+              <label data-l10n-id="fxsync-collections-history">History</label>
+              <details></details>
+            </gaia-switch>
+          </li>
+          <li>
+            <gaia-switch>
```
Comment (evelynhung) [inline apps/settings/elements/fxsync.html]: same here

File: apps/settings/elements/fxsync.html [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+              <label data-l10n-id="fxsync-collections-tabs">Tabs</label>
+              <details></details>
+            </gaia-switch>
+          </li>
+          <li>
+            <gaia-switch>
```
Comment (evelynhung) [inline apps/settings/elements/fxsync.html]: same here.

File: Makefile [hunk 11]
```diff
@@ -87,6 +87,9 @@ GAIA_OPTIMIZE?=0
 GAIA_DEV_PIXELS_PER_PX?=1
 NGA_SERVICE_WORKERS?=0
 
+# For Data Sync
+FIREFOX_SYNC?=0
+
```
Comment (evelynhung) [inline Makefile]: Please write down the actual build command on the bug and/or any Data Sync document, so people who want to try it out could find a correct way to enable it.

**Pull request closed without merging**